### ISSUE TITLE
The schema isn't formatted with wmlindent

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#5709]
The files in data/schema/... are indented with tabs instead of spaces, running wmlindent on them generates around 5000 lines of purely whitespace changes. I suggest we run it and then enable checking the formatting during CI runs.

However, let's not do that right now, as it will generate merge conflicts with @CelticMinstrel's #5663. Reviewing that PR is the only reason I noticed this.

**Comment by CelticMinstrel** [comment]
Just for the record, it won't generate merge conflicts with #5663 - the schema commits in that PR will disappear once I rebase the PR onto master, as they've already been committed to master.

I'm not exactly in favour of this, given that I prefer tabs over spaces everywhere. I do understand that having _consistent_ formatting is helpful, though, and if that's the goal, then schema isn't the only culprit - all the GUI WML also uses tabs.

**Comment by shikadiqueen** [comment]
> all the GUI WML also uses tabs.

That would be because mordante also favoured tabs, plus the C++ and WML are tightly coupled and the C++ uses tabs.

Also, please **DO NOT** this as an invitation to change the GUI2 WML indentation style. I've got plenty of local branches and stashes that rely on it not changing.

**Comment by CelticMinstrel** [comment]
If it were up to me, I'd change the wmlindent style to tabs instead. It's not really up to me, though.

**Comment by shikadiqueen** [comment]
Yeah, that's just not gonna happen simply because it would be bad form for us to go back on the promise we made back then. That's the whole reason why wmlindent doesn't allow you to choose tabs instead of spaces.

**Comment by CelticMinstrel** [comment]
That promise was already broken when mordante decided to use tabs though.

**Comment by shikadiqueen** [comment]
No, it wasn't. GUI2 WML is an implementation detail of the game and people can and should use spaces in their own add-ons.

**Comment by CelticMinstrel** [comment]
Um… add-ons are out-of-scope for wmlindent…

**Comment by shikadiqueen** [comment]
No, they are not. It is designed for both mainline content and add-ons.

**Comment by CelticMinstrel** [comment]
It is not designed for use in add-ons. A design intended to be used by other people would offer options, but it intentionally offers none. This makes sense only if it's intended solely for use on mainline content, where having a lack of options is a good thing (to, as the tool's own documentation says, "prevent style wars" and to ensure that indentation is kept consistent). People can use it for their add-ons if they want, and it has the bare minimum of support for that (ie allowing you to specify arbitrary input files), but it's not the purpose of the tool, and I for one would never recommend it.

**Comment by shikadiqueen** [comment]
wmlindent is designed for use in add-ons as well regardless of what your opinion on the standard formatting style chosen for WML is. It has even received small features and bug fixes involving add-ons and there's even a GUI front-end for it nowadays.

**Comment by cooljeanius** [comment]
Yeah I think it's fair to consider `wmlindent` as being designed for add-ons; I use it on all of mine. There's even a GitHub Action for running it (and `wmllint`) on your own add-ons: https://github.com/czyzby/wesnoth-wml-linter

**Comment by CelticMinstrel** [comment]
If it was really designed for add-ons it would have options, rather than steamrolling one person's preferred coding style onto everyone.

**Comment by irydacea** [comment]
`wmlindent` was expressly designed for using on content meant to be added or already part of mainline, and the original author very intentionally did not include options for this very reason (all this was even stated in the code itself at some point).

**Comment by irydacea** [comment]
With all that said, the whole intention is that add-ons should also follow the same conventions as mainline. It may sound absurd but shortly after its introduction, the idea of *automatically running it on every add-on uploaded to the server* was very considered, and ultimately did not go through simply because the add-on server architecture did not allow for it. The logic was that all add-ons should be ready for a mainline dev to prepare for mainline inclusion and that non-destructive automatic tooling was the way to achieve this.

**Comment by CelticMinstrel** [comment]
> the whole intention is that add-ons should also follow the same conventions as mainline.

That's an utterly absurd intention. About as absurd as suggesting PIP-8 should be used by everyone everywhere.

> the idea of _automatically running it on every add-on uploaded to the server_ was seriously considered

If it's part of the upload process and doesn't touch local files, then I guess it wouldn't really affect me, though it could maybe get in the way of people proposing fixes.

> The logic was that all add-ons should be ready for a mainline dev to prepare for mainline inclusion and that non-destructive automatic tooling was the way to achieve this.

That's certainly not a great logic.

"If you're going to make a mod for my game, you have to be prepared to have me snatch it away and include it with the game."

If that was a real policy I imagine it could turn some people away. (Of course, it would also be attractive to some people, but still.)

**Comment by Elvish-Hunter** [comment]
> the original author very intentionally did not include options for this very reason (all this was even stated in the code itself at some point)

It still is:

> The indent unit is four spaces.  Absence of an option to change this is deliberate; the purpose of this tool is to *prevent* style wars, not encourage them.
